packages/react-log-viewer/src/LogViewer/utils/utils.tsx

```diff
@@ -42,9 +42,45 @@ export const parseConsoleOutput = (data: string) => {
 
 export const escapeString = (inputString: string): string => inputString.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'); // eslint-disable-line
 
-/* eslint-disable no-control-regex */
-export const stripAnsi = (inputString: string): string =>
-  inputString.replace(/[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-PRZcf-nqry=><]/g, '');
+/* eslint-disable-next-line no-control-regex */
+export const ansiRegex = new RegExp(/[\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-PRZcf-nqry=><]/g);
+
+export const isAnsi = (inputString: string) => inputString.match(ansiRegex);
+
+export const stripAnsi = (inputString: string): string => inputString.replace(ansiRegex, '');
+
+export const splitAnsi = (inputString: string): string[] => {
```

Comment:
could this function be simplified by something like
```
` task\x1b[0m\n\x1b[32m>> \x1b[39m0 paths cleaned.\n\n\x1b[4;1mRunning "copy:base"`.split(new RegExp(/([\u001b\u009b][[()#;?]*(?:[0-9]{1,4}(?:;[0-9]{0,4})*)?[0-9A-PRZcf-nqry=><])/g));
```
Note: this is the same regex as `ansiRegex`, i just surrounded it with braces `()`

Reply:
@jschuler I tried and looks like we cannot replace it with the function above. The purpose of the function is to split a string into some format like `[str, ansi, str, ansi, str, ...]`. However, all the ansi escape code will be stripped if we simply call the split function to split it.

Reply:
@jschuler Oh sorry I misunderstood what you mean and didn't notice the `braces`, I will try it again, thanks!

Reply:
It works, thanks for the suggestion!